tools/devctr/Dockerfile

```diff
@@ -0,0 +1,102 @@
+FROM ubuntu:22.04
+
+# TODO: use a multi-stage build to reduce the download size when updating this container.
+# The Rust toolchain layer will get updated most frequently, but we could keep the system
+# dependencies layer intact for much longer.
+
+ARG RUST_TOOLCHAIN="1.66.1"
+ARG TMP_BUILD_DIR=/tmp/build
+ARG FIRECRACKER_SRC_DIR="/firecracker"
+ARG FIRECRACKER_BUILD_DIR="$FIRECRACKER_SRC_DIR/build"
+ARG CARGO_REGISTRY_DIR="$FIRECRACKER_BUILD_DIR/cargo_registry"
+ARG CARGO_GIT_REGISTRY_DIR="$FIRECRACKER_BUILD_DIR/cargo_git_registry"
+ARG DEBIAN_FRONTEND=noninteractive
+# By default we don't provide a poetry.lock file
+ARG POETRY_LOCK_PATH="/dev/null/*"
+ARG ARCH
+
+ENV CARGO_HOME=/usr/local/rust
+ENV RUSTUP_HOME=/usr/local/rust
+ENV PATH="$PATH:$CARGO_HOME/bin"
+ENV LC_ALL=C.UTF-8
+
+# Install system dependencies
+#
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends \
+        # essential build tools
+        gcc make libc-dev binutils-dev libssl-dev \
+        # Needed in order to be able to compile `userfaultfd-sys`.
+        clang \
+        curl \
+        file \
+        git \
+        jq \
+        less \
+        libbfd-dev \
+        # for pandas
+        libbz2-dev \
+        libdw-dev \
+        # for aarch64, but can install in x86_64
+        libfdt-dev \
+        libiberty-dev \
+        libcurl4-openssl-dev \
+        lsof \
+        musl-tools \
+        net-tools iproute2 iperf3 \
+        openssh-client \
+        pkgconf \
+        python3 python3-dev python3-pip \
+        screen tmux \
+        tzdata \
+        tini \
+        # for swagger-cli
+        npm \
+        # for mdl
+        ruby \
+    && rm -rf /var/lib/apt/lists/* \
+    && pip3 install --upgrade pip poetry \
+    && gem install mdl \
+    && npm install -g @apidevtools/swagger-cli
+
+
+COPY tools/devctr /tmp/poetry
+RUN cd /tmp/poetry && \
+    HOME=. POETRY_VIRTUALENVS_CREATE=false poetry install --only main --no-interaction \
+    && rm -rf ~/.cache ~/.npm
+
+# Install the Rust toolchain
+RUN mkdir "$TMP_BUILD_DIR" \
+    && curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain "$RUST_TOOLCHAIN" \
+        && rustup target add $ARCH-unknown-linux-musl \
+        && rustup component add llvm-tools-preview \
+        && cd "$TMP_BUILD_DIR" \
+            && cargo install cargo-audit cargo-deny \
+            && cargo install grcov \
+        && rm -rf "$CARGO_HOME/registry" \
+        && ln -s "$CARGO_REGISTRY_DIR" "$CARGO_HOME/registry" \
+        && rm -rf "$CARGO_HOME/git" \
+        && ln -s "$CARGO_GIT_REGISTRY_DIR" "$CARGO_HOME/git" \
+    && cd / \
+    && rm -rf "$TMP_BUILD_DIR"
```

Comment:
In the previous docker files, we didn't use `$TMP_BUILD_DIR` here. Could you add a comment explaining why we need this here.

Reply:
I think this is what it was when I wrote it originally. Let me see if there is any diff...

Reply:
Ok so I don't see any intermediate packages... looks like `cargo install` [uses a temp dir by default](https://doc.rust-lang.org/cargo/commands/cargo-install.html), so `$TMP_BUILD_DIR` was never needed?

Reply:
I think its fair to condense `cargo install`s into 1 line here.